### PR TITLE
WIP: Use Bytedance's Sonic JSON encoding/json replacement

### PR DESCRIPTION
**Pull request by techknowlogick**
Note: This only works for certain OSes & Arches, so it should not be made the default.

Making this a draft so it doesn't fill up inboxes when I send testing commits to it. If it gets to a good place, then I might consider marking as ready for review.

### CODE REVIEW

**Comment by lunny** [comment]
Please also add update xorm's JSON implementation.

**Comment by wxiaoguang** [comment]
Are there some benchmarks in real production environment to evaluate how much benefit it brings?

There could be a balance between benefit/complexity/maintainability

File: modules/json/sonic.go [hunk 1]
```diff
@@ -0,0 +1,177 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
```
Comment (yardenshoham) [inline modules/json/sonic.go]:
```suggestion
// Copyright 2023 The Gitea Authors. All rights reserved.
```

**Comment by techknowlogick** [comment]
I'm not so sure about this PR, seems to add significant build time. jsoniter is doing its job and I don't see it coming up in any flamegraphs. I'm going to close this. It could maybe be revisited in the future, or if someone sees a glaring mistake I've made please feel free to tell me :)

**Pull request closed without merging**